doc/manual/source/developer_guide/ModificationIntro.rst

```diff
@@ -37,152 +37,226 @@ mailing list:
+
+   .. note::
+     Note that the above approach uses HTTPS as the transfer protocol
+     between your machine and GitHub.  If you prefer to use SSH - or
+     perhaps you're behind a proxy that doesn't play well with SSL via
+     HTTPS - you may want to set up an `SSH key
+     <https://help.github.com/articles/connecting-to-github-with-ssh/>`__
+     on GitHub.  Then, you use
+     the syntax ``ssh://git@github.com/<USER>/enzo_dev``, or equivalent, in
+     place of ``https://github.com/<USER>/enzo_dev`` in git commands.
+     For consistency, all commands we list in this document will use the HTTPS
+     protocol.
+
+#. Issue a pull request by going to the main repository and clicking on the
+   green button that says **Compare & pull request**. This will open up a page
+   that will allow you to enter a description of the changes to be merged. Once
+   submitted, a series of automated tests will run and their status will be
+   reported on the pull request page.
+
+During the course of your pull request you may be asked to make changes.  These
+changes may be related to style issues, correctness issues, or requesting
+tests.  The process for responding to pull request code review is relatively
+straightforward.
+
+#. Make requested changes, or leave a comment on the pull request page on
+   GitHub indicating why you don't think they should be made.
+#. Commit those changes to your local repository.
+#. Push the changes to your fork::
+
+      $ git push origin <branch name>
+
+#. Your pull request will be automatically updated.
+
+Once your pull request has been accepted, you can safely delete your
+branch::
+
+      $ git branch --delete <branch name>
+
+Updating Your Branch
+++++++++++++++++++++
+
+If your branch or pull request has been open for some time, it may be useful
+to keep it up to date with the latest changes from the main repository. This
+can be done by `rebasing your changes <https://git-scm.com/docs/git-rebase>`__.
+Before doing this, you will need to be able to pull the latest changes from
+the main repository.
+
+#. Add the main repository as a remote::
+
+      $ git remote add enzo_dev https://github.com/enzo-project/enzo-dev
+
+   You can verify that it has been added by doing ``git remote -v``. This
+   only needs to be done once.
+
+#. Go back to the master branch and pull the changes::
+
+      $ git checkout master
+      $ git pull enzo_dev master
+
+#. Return to your branch and rebase your changes onto the head of the master
+   branch::
+
+      $ git checkout <branch name>
+      $ git rebase master
+
+This should go smoothly unless changes have been made to the same lines in
+the source, in which case you will need to fix conflicts. After rebasing,
+you will get an error when trying to push your branch to your fork. This is
+because you have changed the order of commits and git does not like that.
+In this case, you will need to add "-f" to your push command to force
+the changes to be accepted.::
+
```

Comment:
Perhaps you could add a brief note hearing saying that if you envision a decent number of conflicts, it may be better to do `get merge master` to do a merge commit with master rather than rebasing.